### ISSUE TITLE
Access violation when running Autoroute in PCB view

### ISSUE THREAD

**Issue by zhuowei** [fritzing/fritzing-app#3048]
I tried autorouting a circuit for the first time, and Fritzing crashed.

I'm using fritzing.0.9.1b.64.pc on Windows 8.1.

Crash log from Windbg:

```
Microsoft (R) Windows Debugger Version 6.3.9600.16384 AMD64
Copyright (c) Microsoft Corporation. All rights reserved.

*** wait with pending attach

************* Symbol Path validation summary **************
Response                         Time (ms)     Location
Deferred                                       srv*d:\symbols*http://msdl.microsoft.com/download/symbols
Symbol search path is: srv*d:\symbols*http://msdl.microsoft.com/download/symbols
Executable search path is: 
ModLoad: 00007ff6`dfb10000 00007ff6`e02c4000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Fritzing.exe
ModLoad: 00007ffd`88020000 00007ffd`881cc000   C:\WINDOWS\SYSTEM32\ntdll.dll
ModLoad: 00007ffd`85ee0000 00007ffd`8601e000   C:\WINDOWS\system32\KERNEL32.DLL
ModLoad: 00007ffd`854d0000 00007ffd`855e5000   C:\WINDOWS\system32\KERNELBASE.dll
ModLoad: 00007ffd`66a20000 00007ffd`66a68000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5PrintSupport.dll
ModLoad: 00000000`70d70000 00000000`70db3000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Svg.dll
ModLoad: 00000000`70860000 00000000`70d6c000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Widgets.dll
ModLoad: 00007ffd`72600000 00007ffd`72615000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5SerialPort.dll
ModLoad: 00000000`70820000 00000000`70854000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Sql.dll
ModLoad: 00000000`70720000 00000000`7081d000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Network.dll
ModLoad: 00000000`706e0000 00000000`70716000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Xml.dll
ModLoad: 00007ffd`58320000 00007ffd`586c9000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Gui.dll
ModLoad: 00000000`70240000 00000000`706de000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Core.dll
ModLoad: 00007ffd`60f10000 00007ffd`60fb7000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\MSVCP110.dll
ModLoad: 00007ffd`60e30000 00007ffd`60f04000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\MSVCR110.dll
ModLoad: 00007ffd`87970000 00007ffd`87ac1000   C:\WINDOWS\system32\GDI32.dll
ModLoad: 00007ffd`87dd0000 00007ffd`87f47000   C:\WINDOWS\system32\USER32.dll
ModLoad: 00007ffd`7dff0000 00007ffd`7e072000   C:\WINDOWS\SYSTEM32\WINSPOOL.DRV
ModLoad: 00007ffd`87f50000 00007ffd`88006000   C:\WINDOWS\system32\COMDLG32.dll
ModLoad: 00007ffd`86040000 00007ffd`87558000   C:\WINDOWS\system32\SHELL32.dll
ModLoad: 00007ffd`87640000 00007ffd`8781a000   C:\WINDOWS\system32\SETUPAPI.dll
ModLoad: 00007ffd`87b70000 00007ffd`87c1a000   C:\WINDOWS\system32\ADVAPI32.dll
ModLoad: 00007ffd`859f0000 00007ffd`85a4a000   C:\WINDOWS\system32\WS2_32.dll
ModLoad: 00007ffd`84850000 00007ffd`848f4000   C:\WINDOWS\SYSTEM32\DNSAPI.dll
ModLoad: 00007ffd`85240000 00007ffd`8541f000   C:\WINDOWS\system32\CRYPT32.dll
ModLoad: 00007ffd`59d90000 00007ffd`59e66000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\libGLESv2.dll
ModLoad: 00007ffd`87c30000 00007ffd`87dc4000   C:\WINDOWS\system32\ole32.dll
ModLoad: 00000000`70010000 00000000`70238000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\icuin51.dll
ModLoad: 00000000`6fe60000 00000000`70004000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\icuuc51.dll
ModLoad: 00007ffd`85e20000 00007ffd`85eca000   C:\WINDOWS\system32\msvcrt.dll
ModLoad: 00007ffd`85760000 00007ffd`857b4000   C:\WINDOWS\system32\SHLWAPI.dll
ModLoad: 00007ffd`6aa40000 00007ffd`6aae4000   C:\WINDOWS\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_5.82.9600.17415_none_34aa3313958e7a52\COMCTL32.dll
ModLoad: 00007ffd`85a50000 00007ffd`85c61000   C:\WINDOWS\SYSTEM32\combase.dll
ModLoad: 00007ffd`855f0000 00007ffd`8563f000   C:\WINDOWS\system32\CFGMGR32.dll
ModLoad: 00007ffd`87820000 00007ffd`87961000   C:\WINDOWS\system32\RPCRT4.dll
ModLoad: 00007ffd`87b10000 00007ffd`87b69000   C:\WINDOWS\SYSTEM32\sechost.dll
ModLoad: 00007ffd`87c20000 00007ffd`87c29000   C:\WINDOWS\system32\NSI.dll
ModLoad: 00007ffd`85220000 00007ffd`85231000   C:\WINDOWS\system32\MSASN1.dll
ModLoad: 00007ffd`740f0000 00007ffd`74311000   C:\WINDOWS\SYSTEM32\d3d9.dll
ModLoad: 00000000`6e910000 00000000`6fe5a000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\icudt51.dll
ModLoad: 00007ffd`7c980000 00007ffd`7c98a000   C:\WINDOWS\SYSTEM32\VERSION.dll
ModLoad: 00007ffd`82f70000 00007ffd`82f91000   C:\WINDOWS\SYSTEM32\dwmapi.dll
ModLoad: 00007ffd`82da0000 00007ffd`82e52000   C:\WINDOWS\SYSTEM32\SHCORE.DLL
ModLoad: 00007ffd`87ad0000 00007ffd`87b06000   C:\WINDOWS\system32\IMM32.DLL
ModLoad: 00007ffd`857c0000 00007ffd`85913000   C:\WINDOWS\system32\MSCTF.dll
ModLoad: 00007ffd`850b0000 00007ffd`850bb000   C:\WINDOWS\SYSTEM32\CRYPTBASE.DLL
ModLoad: 00007ffd`85040000 00007ffd`850a3000   C:\WINDOWS\SYSTEM32\bcryptPrimitives.dll
ModLoad: 00007ffd`58210000 00007ffd`5831c000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\platforms\qwindows.dll
ModLoad: 00007ffd`7ddd0000 00007ffd`7ddf2000   C:\WINDOWS\SYSTEM32\WINMM.dll
ModLoad: 00007ffd`87570000 00007ffd`87631000   C:\WINDOWS\system32\OLEAUT32.dll
ModLoad: 00007ffd`6a330000 00007ffd`6a342000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\libEGL.dll
ModLoad: 00007ffd`7dd90000 00007ffd`7ddba000   C:\WINDOWS\SYSTEM32\WINMMBASE.dll
ModLoad: 00007ffd`83f60000 00007ffd`83f88000   C:\WINDOWS\SYSTEM32\DEVOBJ.dll
ModLoad: 00007ffd`83fd0000 00007ffd`83fdb000   C:\WINDOWS\SYSTEM32\kernel.appcore.dll
ModLoad: 00007ffd`83de0000 00007ffd`83f09000   C:\WINDOWS\system32\uxtheme.dll
ModLoad: 00007ffd`7d410000 00007ffd`7d4bb000   C:\Program Files\Common Files\microsoft shared\ink\tiptsf.dll
ModLoad: 00007ffd`84790000 00007ffd`847b1000   C:\WINDOWS\system32\userenv.dll
ModLoad: 00007ffd`85170000 00007ffd`85185000   C:\WINDOWS\SYSTEM32\profapi.dll
ModLoad: 00007ffd`65a80000 00007ffd`65abe000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\lib\imageformats\qjpeg.dll
ModLoad: 00007ffd`80a10000 00007ffd`80a7a000   C:\WINDOWS\SYSTEM32\oleacc.dll
ModLoad: 00007ffd`58140000 00007ffd`58207000   C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\lib\sqldrivers\qsqlite.dll
ModLoad: 00000001`80000000 00000001`80061000   C:\Program Files\Intel\iCLS Client\ssleay32.dll
ModLoad: 000000a1`2a720000 000000a1`2a8b4000   C:\Program Files\Intel\iCLS Client\LIBEAY32.dll
ModLoad: 00007ffd`84ab0000 00007ffd`84ad0000   C:\WINDOWS\SYSTEM32\CRYPTSP.dll
ModLoad: 00007ffd`84680000 00007ffd`846b6000   C:\WINDOWS\system32\rsaenh.dll
ModLoad: 00007ffd`84d00000 00007ffd`84d26000   C:\WINDOWS\SYSTEM32\bcrypt.dll
ModLoad: 00007ffd`84a50000 00007ffd`84aa9000   C:\WINDOWS\system32\mswsock.dll
ModLoad: 00007ffd`764a0000 00007ffd`764aa000   C:\Windows\System32\rasadhlp.dll
ModLoad: 00007ffd`856a0000 00007ffd`85756000   C:\WINDOWS\SYSTEM32\clbcatq.dll
ModLoad: 00007ffd`80600000 00007ffd`8062a000   C:\WINDOWS\SYSTEM32\IPHLPAPI.DLL
ModLoad: 00007ffd`805e0000 00007ffd`805ea000   C:\WINDOWS\SYSTEM32\WINNSI.DLL
ModLoad: 00007ffd`71960000 00007ffd`71973000   C:\Program Files\Windows Journal\NBMapTIP.dll
ModLoad: 00007ffd`6bd40000 00007ffd`6be93000   C:\WINDOWS\SYSTEM32\MFC42u.dll
ModLoad: 00007ffd`6bc70000 00007ffd`6bd32000   C:\WINDOWS\SYSTEM32\ODBC32.dll
ModLoad: 00007ffd`85640000 00007ffd`85691000   C:\WINDOWS\system32\WINTRUST.dll
ModLoad: 00007ffd`7d4c0000 00007ffd`7d524000   C:\WINDOWS\system32\Ninput.dll
ModLoad: 00007ffd`81830000 00007ffd`81aab000   C:\WINDOWS\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.9600.17415_none_6240486fecbd8abb\comctl32.DLL
ModLoad: 00007ffd`82a20000 00007ffd`82bce000   C:\WINDOWS\SYSTEM32\WindowsCodecs.dll
(1d68.2034): Access violation - code c0000005 (!!! second chance !!!)
*** WARNING: Unable to verify checksum for C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Fritzing.exe
*** ERROR: Module load completed but symbols could not be loaded for C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Fritzing.exe
Fritzing+0x1cfb97:
00007ff6`dfcdfb97 48890cd0        mov     qword ptr [rax+rdx*8],rcx ds:000000a1`40d40d08=????????????????
0:000> k
Child-SP          RetAddr           Call Site
000000a1`25a499f0 00007ff6`dfce7d6c Fritzing+0x1cfb97
000000a1`25a49a80 00007ff6`dfce4b5b Fritzing+0x1d7d6c
000000a1`25a49dd0 00007ff6`dfcea92e Fritzing+0x1d4b5b
000000a1`25a4a110 00007ff6`dfcecf31 Fritzing+0x1da92e
000000a1`25a4a610 00007ff6`dfb6c854 Fritzing+0x1dcf31
000000a1`25a4a9e0 00007ff6`dfdb4a8a Fritzing+0x5c854
*** ERROR: Symbol file could not be found.  Defaulted to export symbols for C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Core.dll - 
000000a1`25a4aaf0 00000000`7042d131 Fritzing+0x2a4a8a
*** ERROR: Symbol file could not be found.  Defaulted to export symbols for C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Widgets.dll - 
000000a1`25a4ab60 00000000`70866471 Qt5Core!QMetaObject::activate+0x5b1
000000a1`25a4ac70 00000000`70943856 Qt5Widgets!QAction::activate+0xf1
000000a1`25a4acc0 00000000`70944768 Qt5Widgets!QAbstractButton::click+0x136
000000a1`25a4ad00 00000000`709f593f Qt5Widgets!QAbstractButton::mouseReleaseEvent+0x88
000000a1`25a4ad30 00000000`70899d4f Qt5Widgets!QToolButton::mouseReleaseEvent+0xf
000000a1`25a4ad60 00000000`70874366 Qt5Widgets!QWidget::event+0xdf
000000a1`25a4aef0 00000000`7087251b Qt5Widgets!QApplicationPrivate::notify_helper+0xf6
000000a1`25a4af20 00007ff6`dfb25384 Qt5Widgets!QApplication::notify+0x8db
000000a1`25a4b570 00000000`7040d3da Fritzing+0x15384
000000a1`25a4b630 00000000`70875392 Qt5Core!QCoreApplication::notifyInternal+0x6a
000000a1`25a4b690 00000000`708c0348 Qt5Widgets!QApplicationPrivate::sendMouseEvent+0x372
000000a1`25a4b7a0 00000000`708beb5e Qt5Widgets!QWidget::windowHandle+0x1c18
000000a1`25a4b980 00000000`70874366 Qt5Widgets!QWidget::windowHandle+0x42e
000000a1`25a4b9e0 00000000`70872f06 Qt5Widgets!QApplicationPrivate::notify_helper+0xf6
000000a1`25a4ba10 00007ff6`dfb25384 Qt5Widgets!QApplication::notify+0x12c6
000000a1`25a4c060 00000000`7040d3da Fritzing+0x15384
*** ERROR: Symbol file could not be found.  Defaulted to export symbols for C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\Qt5Gui.dll - 
000000a1`25a4c120 00007ffd`5834c4d7 Qt5Core!QCoreApplication::notifyInternal+0x6a
000000a1`25a4c180 00007ffd`5834e6d7 Qt5Gui!QGuiApplicationPrivate::processMouseEvent+0x4d7
000000a1`25a4c4a0 00007ffd`5833cefe Qt5Gui!QGuiApplicationPrivate::processWindowSystemEvent+0x137
000000a1`25a4c4f0 00000000`7044d995 Qt5Gui!QWindowSystemInterface::sendWindowSystemEvents+0x11e
000000a1`25a4c530 00007ffd`87dd250d Qt5Core!QEventDispatcherWin32::processEvents+0xd95
000000a1`25a4c620 00007ffd`87dd2367 USER32!UserCallWinProcCheckWow+0x149
000000a1`25a4c6f0 00000000`7044d0cc USER32!DispatchMessageWorker+0x1a7
*** WARNING: Unable to verify checksum for C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\platforms\qwindows.dll
*** ERROR: Symbol file could not be found.  Defaulted to export symbols for C:\Users\zhuowei\Documents\winprogress\fritzing.0.9.1b.64.pc\platforms\qwindows.dll - 
000000a1`25a4c770 00007ffd`5823031d Qt5Core!QEventDispatcherWin32::processEvents+0x4cc
000000a1`25a4fad0 00000000`7040a5b1 qwindows!qt_plugin_query_metadata+0x1e9ed
000000a1`25a4fb40 00000000`7040c647 Qt5Core!QEventLoop::exec+0x1c1
000000a1`25a4fbc0 00007ff6`dfb31bd7 Qt5Core!QCoreApplication::exec+0x187
000000a1`25a4fc30 00007ff6`dfdca5ce Fritzing+0x21bd7
000000a1`25a4fca0 00007ff6`dfdc9f60 Fritzing+0x2ba5ce
000000a1`25a4fd10 00007ffd`85ee13d2 Fritzing+0x2b9f60
000000a1`25a4fd50 00007ffd`880903c4 KERNEL32!BaseThreadInitThunk+0x22
000000a1`25a4fd80 00000000`00000000 ntdll!RtlUserThreadStart+0x34
0:000> r
rax=000000a140d41020 rbx=0000000000000028 rcx=fffffffffffffffd
rdx=ffffffffffffff9d rsi=0000000000000028 rdi=0000000000000048
rip=00007ff6dfcdfb97 rsp=000000a125a499f0 rbp=000000a12f024c60
 r8=fffffffffffffef4  r9=00000000fffffee0 r10=ffffffffffffffff
r11=000000000000000f r12=000000000000002d r13=000000a125a49ae8
r14=ffffffffffffffff r15=000000a134423790
iopl=0         nv up ei ng nz na pe nc
cs=0033  ss=002b  ds=002b  es=002b  fs=0053  gs=002b             efl=00010280
Fritzing+0x1cfb97:
00007ff6`dfcdfb97 48890cd0        mov     qword ptr [rax+rdx*8],rcx ds:000000a1`40d40d08=????????????????
```

**Comment by zhuowei** [comment]
I uploaded the file that causes the crash: https://drive.google.com/file/d/0B-fqnmBSesbtWFh4OFZkd0I0XzA/view?usp=sharing

**Comment by KjellMorgenstern** [comment]
Copied files from google docs to github:
[motordriver.fzz.zip](https://github.com/fritzing/fritzing-app/files/3407326/motordriver.fzz.zip)

**Comment by failiz** [comment]
I could not replicate this in 0.9.4.

**Comment by KjellMorgenstern** [comment]
Several crashes were fixed for 0.9.4 I had initially even removed the autoroutes, since it was easy to crash. But several bug were fixe, and we could keep the autorouter in.
